app/src/main/java/org/wikipedia/views/imageservice/WhiteBackgroundTransformation.kt

```diff
@@ -1,28 +1,29 @@
 package org.wikipedia.views.imageservice
 
-import android.graphics.Matrix
-import androidx.core.graphics.createBitmap
+import android.graphics.Paint
+import android.graphics.PorterDuff
+import android.graphics.PorterDuffXfermode
+import androidx.core.graphics.applyCanvas
 import coil3.Bitmap
 import coil3.size.Size
 import coil3.transform.Transformation
-import org.wikipedia.extensions.applyMatrixWithBackground
-import org.wikipedia.extensions.maybeDimImage
 
 class WhiteBackgroundTransformation : Transformation(), ImageTransformer {
-    override val cacheKey: String
-        get() = "WhiteBackgroundTransformation"
+    override val cacheKey = "WhiteBackgroundTransformation"
 
     override suspend fun transform(input: Bitmap, size: Size): Bitmap {
-        val result = if (input.hasAlpha()) {
-            val newBitmap = createBitmap(
-                input.width,
-                input.height,
-                input.config ?: android.graphics.Bitmap.Config.ARGB_8888
-            )
-            newBitmap.applyMatrixWithBackground(input, Matrix())
-            newBitmap
-        } else input
-
-        return result.maybeDimImage()
+        if (input.hasAlpha()) {
+            val result = if (input.isMutable) input else input.copy(input.config ?: android.graphics.Bitmap.Config.ARGB_8888, true)
+            val paint = Paint().apply {
+                xfermode = PorterDuffXfermode(PorterDuff.Mode.DST_OVER)
+                color = android.graphics.Color.WHITE
+            }
+            result.applyCanvas {
+                drawRect(0f, 0f, result.width.toFloat(), result.height.toFloat(), paint)
```

Comment:
Performance improvement: For drawing a WHITE background on a bitmap, we can make a single call to `drawRect` with `DST_OVER`, instead of drawing a rect and then drawing the whole bitmap over it.